src/core/utils/building/internal-build.ts

```diff
@@ -104,7 +104,28 @@ async function combineAnalysisStats(): Promise<void> {
     ],
     { cwd: wxt.config.root, stdio: 'inherit' },
   );
-  await Promise.all(absolutePaths.map((statsFile) => fs.remove(statsFile)));
+
+  if (wxt.config?.analysis?.keepArtifacts) {
+    await Promise.all(
+      absolutePaths.map((statsFile) => {
+        const outputDir = path.dirname(
+          wxt.config?.analysis?.outputFile ?? wxt.config.outDir,
+        );
+
+        // Rename stats*.json artifacts based on outputFile filename
+        const cleanedStatsFilename = path
+          .parse(statsFile)
+          .name.replace('stats-', '');
+        const artifactDestinationPath = `${outputDir}/${
+          path.parse(wxt.config?.analysis?.outputFile).name
+        }_${cleanedStatsFilename}.json`;
+
+        fs.move(statsFile, artifactDestinationPath, { overwrite: true });
```

Comment:
Instead of moving files, can we change the filename we write them to in the first place?

https://github.com/wxt-dev/wxt/blob/2b374b98d281521707f1b79e87daf8b931dedd46/src/core/builders/vite/plugins/bundleAnalysis.ts#L13